raster/src/main/scala/geotrellis/raster/interpolation/InverseDistanceWeighting.scala

```diff
@@ -0,0 +1,139 @@
+package geotrellis.raster.interpolation
+
+import geotrellis.raster._
+import geotrellis.raster.mapalgebra.focal.Circle
+import geotrellis.vector.{Extent, PointFeature, SpatialIndex}
+import spire.syntax.cfor._
+
+object InverseDistanceWeighting {
+  /**
+    * Compute an Inverse Distance Weighting raster over the given
+    * extent from the given set known-points.  Please see
+    * https://en.wikipedia.org/wiki/Inverse_distance_weighting for
+    * more details.
+    *
+    * @param   points    A collection of known-points
+    * @param   rasterExtent The study area
+    * @param   radius    Interpolation radius in coordinate unit
+    * @param   equalWeightRadius If any points lie at that distance from a cell, the interpolation result for it would
+    *                            be the mean of those points
+    * @param   cellType  Interpolation radius in coordinate unit
+    * @param   f         Function to be applied before storing interpolation values
```

Comment:
To be clear this exists because when setting a double to an int based raster the default behavior is to truncate. I can see why that would not always be desirable. The name should reflect something about when it's used `onSet` should do it.